labs/src/main/scala/org/scalalabs/basic/lab02/CollectionExercise.scala

```diff
@@ -50,7 +79,7 @@ object CollectionExercise02 {
    * using a functional approach.
    */
   def groupAdultsPerAgeGroup(persons: Seq[Person]): Map[Int, Seq[Person]] = {
-    error("fix me")
+    persons.filter((x:Person) => x.age >= 18).groupBy { (x:Person) => 10 * (x.age/10) }
   }
 }
```

Comment:
This method doesn't sort the people by name, which is a feature of the Java method. If you wanted to add this functionality and decrease the overall size of the method, you could consider using  placeholders (_.age >= 18) like the sample solution.
Also, do you need to explicitly define x as a Person in this method? I think Scala will automatically infer that type, though I'm not sure.